### DISCUSSION TITLE
Compiler "hints"

### DISCUSSION THREAD

**Discussion by mansellan** [twinbasic/twinbasic#1335] [Ideas]
We now have a bunch of compiler warnings for things that look suspicious. I wonder if we should also have compiler hints, for things that look suboptimal to the compiler. To avoid consternation for new users when importing VBx code, these would default OFF in the project settings, and be opt-in.

Thoughts?

**Comment by dmrvb** [comment]
Its a good idea and I think this has been talked about here before, but I would not like this to have any priority at the moment, there is still lots of essential GUI stuff to get working first.

**Reply by wqweto** [reply]
There is a roadmap so the big chunk of work for Q4 is already panned. The “essential GUI stuff” can reach only to bugfixes of existing components.

**Reply by fafalone** [reply]
???

Finishing the GUI controls and adding the UserControls was scheduled as Q3, so I think that comes before the originally scheduled Q4 work.

I don't think it would make much sense to move these things past all roadmap work... start working on multi-platform stuff and massive addins before basic drag drop support or ScaleMode/ScaleX/etc matching VB?

**Reply by wqweto** [reply]
Yes, stick to the roadmap but there inevitably are going to be emergencies.

I’m positive the low-hanging fruits have been collected so if anything looks simple (like ScaleMode) it’s probably not enough simple and not enough of a blocker e.g. Load statement on forms can be stubbed by a Load procedure which does nothing.

**Reply by fafalone** [reply]
Not suggesting the missing stuff is simple, just that there's a lot more than bugfixes left with GUI controls/forms and that it wouldn't make sense to consider that as not part of the 'finish basic controls' work, but instead as something entirely new that would come after all the currently scheduled things on the roadmap. 

I.e. things like non-modal forms, the drag/drop stuff, data binding in the basic controls, and a few other things, are too significant for ability to open/run VB6 projects to be lower priorities than cross-platform support, mixed bitness controls, etc, that wouldn't be started until a year from now, after those.

Though while some of the really obscure stuff might be able to wait, that should be weighed carefully... where does '100% compatibility' fall on the roadmap... should those things come when it's 99% and the the remaining 1% waits? Like I'd bet 99% of us have never used the DDE Link stuff other than to see what it is; but some people do, and it's not 100% compatible without it.

**Comment by dmrvb** [comment]
I believe that getting something that is VB compatible should be top priority. This means importing and compiling the vast majority of VB6/VBA code. Its still a little way off, especially on the GUI side. New functionality is great, and if Wayne is working on something and can easily add a few extra features at the same time then that makes sense, but major new functionality should only come after a V1 VB6/VBA release.

**Reply by rexxitall** [reply]
Sometimes it is nessecary to think a step further to avoid big changes in the design later. They all do a great job . Not always the straight way is the fastest.:)
